Firestore/core/src/util/byte_stream_apple.mm

```diff
@@ -64,24 +61,31 @@
   return read_result;
 }
 
-StreamReadResult ByteStreamNSInputStream::Read(size_t max_length) {
-  auto read = ReadToBuffer(max_length);
+StreamReadResult ByteStreamApple::Read(size_t max_length) {
+  // Serve from buffer_
+  if (buffer_.size() >= max_length) {
```

Comment:
Can you add a test that would fail without this addition?

Reply:
Done.

Reply:
By the way, which test would fail otherwise?